### PR TITLE
Fixed issue #932

### PR DESCRIPTION
**Pull request by pszalanski**
The regular expression to parse options has been extended to allow the notation including encoding.

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
this needs tests,
at first glance this very likely changes external behavior wrt string types

**Comment by pszalanski** [comment]
You are right. Apparently, this breaks the CI build, so this needs some more work. I'll fix the test for now and implement a test for the new behavior afterwards.

**Comment by pszalanski** [comment]
Okay the build is fixed. However, it looks like the latest uwsgi release from yesterday breaks compatibility with Python 2.6. So this fail is not my fault :smile:

**Comment by jeffwidman** [comment]
Fixes #932 (it says it in the title, but not the comments, so Github isn't cross-linking)

**Comment by pszalanski** [comment]
Any plans on merging the change? We at our company are looking forward to getting our hands on the fix.

**Comment by untitaker** [comment]
Will take a look, this slipped through the cracks, thanks.

File: tests/test_http.py [hunk 1]
```diff
@@ -265,6 +265,20 @@ def test_parse_options_header(self):
                                          '        '
                                          'text/x-dvi; q=0.8, text/x-c') == \
             ('text/plain', {'q': '0.5'})
+        # Issue #932
+        assert http.parse_options_header(
+            'form-data; '
```
Comment (pszalanski) [inline tests/test_http.py]: It doesn't have to be, because option headers can be anything, e.g. a mimetype or a filename.

**Comment by untitaker** [comment]
Thanks!

The regex looks a bit scary right now, I'll have a look at how to improve readability.

**Comment by suzaku** [comment]
+1

**Pull request merged**